toplevel/coqtop.ml

```diff
@@ -141,6 +141,7 @@ let ltac_debug_answer = let open DebugHook.Answer in function
       Format.fprintf !Topfmt.err_ft "@[%a@]@\n%!" Pp.pp_with o
     | Init ->
       Format.fprintf !Topfmt.err_ft "@[%a@]@\n%!" Pp.pp_with (str "Init")
+    | Vars _ -> ()  (* todo *)
```

Comment:
`Vars` has a value of type `(string * Pp.t) list`.  Is there a simple way to marshall this using the existing type definition for the `Db_vars` message in `Xmlprotocol`?  If I need to write new marshalling code, what should that look like?

Reply:
Should be just `of_pair of_string of_pp` ? Actually this marshalling code is super annoying, and hindering your work big time.

It wouldn't take long to remove xmlprotocol at all and have `ppx_deriving_yojson` generate the file automatically; tho users of the raw protocol would have to adapt.

Reply:
Close, should be `let of_vars vars = of_list (of_pair of_string of_pp) vars`.  But, as usual, it took a fair bit of time to reverse engineer how to do this.

I want to keep xmlprotocol for this PR, but I'm open to considering changes a bit later.  There are a bunch of things in CoqIDE that would benefit from simplification and refactoring.

Reply:
However the conversion belongs elsewhere; this particular line of code is for `coqtop`.  This should just be a `failwith`.  I'm not planning to add stack and variable display to the coqtop version.  Indeed, I'd like to drop the support in `coqtop` since it's so much less usable.  But not today.